src/sagemaker/serve/utils/tuning.py

```diff
@@ -216,3 +264,71 @@ def _more_performant(best_tuned_configuration: list, tuned_configuration: list)
             return True
         return False
     return tuned_avg_latency <= best_avg_latency
+
+
+def _more_performant_benchmark(
+    best_tuned_configuration: dict, current_tuned_configuration: dict
+) -> dict:
+    """Returns the configuration with the lowest latency"""
+    if best_tuned_configuration is None:
+        return current_tuned_configuration
+
+    best_avg_latency = best_tuned_configuration["AGV_LATENCY"]
+    current_tuned_avg_latency = current_tuned_configuration["AGV_LATENCY"]
+    best_standard_deviation = best_tuned_configuration["STD_DEVIATION"]
+    current_tuned_standard_deviation = current_tuned_configuration["STD_DEVIATION"]
+
+    if _within_margins(MARGIN, 5, current_tuned_avg_latency, best_avg_latency):
+        if current_tuned_standard_deviation <= best_standard_deviation:
+            return current_tuned_configuration
+        return best_tuned_configuration
+
+    if current_tuned_avg_latency <= best_avg_latency:
+        return current_tuned_configuration
+    return best_tuned_configuration
+
+
+def _run_serial_and_concurrent_benchmarks(pysdk_model, sample_input, max_tuning_duration) -> dict:
+    """Run the benchmarks"""
+    predictor = pysdk_model.deploy(model_data_download_timeout=max_tuning_duration)
+
+    avg_latency, p90, avg_tokens_per_second = _serial_benchmark(predictor, sample_input)
+    throughput_per_second, standard_deviation = _concurrent_benchmark(predictor, sample_input)
+
+    tested_env = copy.deepcopy(pysdk_model.env)
+    logger.info(
+        "Average latency: %s, throughput/s: %s for configuration: %s",
+        avg_latency,
+        throughput_per_second,
+        tested_env,
+    )
+
+    return {
+        "AVG_LATENCY": avg_latency,
+        "TESTED_ENV": tested_env,
+        "P90": p90,
+        "AVG_TOKENS_PER_SECOND": avg_tokens_per_second,
+        "THROUGHPUT_PER_SECOND": throughput_per_second,
+        "STD_DEVIATION": standard_deviation,
+    }
+
+
+def sharded_supported(model_id: str, config_dict: dict) -> bool:
+    """Check if sharded is supported for this ``Model``"""
+    model_type = config_dict.get("model_type", None)
+
+    if model_type is None:
+        return False
+
+    if model_id.startswith("facebook/galactica"):
```

Comment:
Why are we hardcoding this value? Can't it mapped to a model_type?

Reply:
See server code: https://github.com/huggingface/text-generation-inference/blob/main/server/text_generation_server/models/__init__.py

Trying to keep it simple.